### PR TITLE
OCSADV-443: BagsResult implementation

### PR DESCRIPTION
**Pull request by sraaphorst**
Representation of Bags target in `GuideProbeTargets` changed from `Option<SPTarget>` to multi-valued `BagsResult` to allow for three states as required for BAGS feedback.

### CODE REVIEW

File: bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/BagsResult.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import edu.gemini.shared.util.immutable.{Option => GOption}
+import edu.gemini.spModel.obs.context.ObsContext
+import edu.gemini.spModel.pio.{Pio, PioFactory, ParamSet}
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.rich.shared.immutable._
+
+sealed trait BagsResult extends Cloneable {
+  val id: String
+  val observationHash: String = ""
```
Comment (cquiroz) [inline bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/BagsResult.scala]: What's the role of `observationHash`? It doesn't seem to be overidden on the derived classes but it is used in equals

Comment (sraaphorst) [inline bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/BagsResult.scala]: Sorry. My bad. That was supposed to have been removed from this. I'm going to do that and push another commit.

File: bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/BagsResult.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import edu.gemini.spModel.rich.shared.immutable._
+
+sealed trait BagsResult extends Cloneable {
+  val id: String
+  val target: Option[SPTarget] = None
+  def targetAsJava: GOption[SPTarget] = target.asGeminiOpt
+  override def clone: BagsResult = this
+
+  override def equals(obj: scala.Any): Boolean = obj match {
+    case other: BagsResult => target.equals(other.target)
```
Comment (cquiroz) [inline bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/BagsResult.scala]: This means that e.g. `NoSearchPerformed == NoTargetFound` is true

Comment (sraaphorst) [inline bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/BagsResult.scala]: Hmmm... good point. Not sure what the easiest way is to fix this with case classes of a sealed trait where I want to compare data members.
I can compare `id`s or something more elaborate.

Comment (cquiroz) [inline bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/BagsResult.scala]: Could you move the equals to only `WithTarget`?

Comment (sraaphorst) [inline bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/BagsResult.scala]: Of course. Ha! Thanks!

Comment (cquiroz) [inline bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/BagsResult.scala]: Actually the default `equals` of the case class should suffice right?

Comment (sraaphorst) [inline bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/BagsResult.scala]: Hmmm... now that you mention it, my guess is that it will, but I could not quite remember how equality is implemented.

I'm also not sure if, as this progresses, we'll want to compare the SPTargets or the underlying ITargets to avoid comparing watchers, but I'm going to take it out completely for now until the need to do something fancier than the default becomes apparent.

**Comment by cquiroz** [comment]
Looks fine :+1:
